### ISSUE TITLE
Rule request: Floating point numbers should not be tested for equality (from sonar rule RSPEC-1244)

### ISSUE THREAD

**Issue by daara-s** [astral-sh/ruff#12615]
Keywords: RSPEC-1244, float, sonar

`RSPEC-1244` from [sonar rules](https://rules.sonarsource.com/python/RSPEC-1244/) checks whether equality checks are made on floating point numbers, which often leads to non-deterministic results in tests.

This rule forms part of Sonar's Python scanner checks, which I've seen is [open as a feature request](https://github.com/astral-sh/ruff/issues/4935). 

Would it be possible to add this individual rule either as a one-off or as a starting point in porting sonar's rules over?

**Comment by MichaReiser** [comment]
Makes sense to me. The one big question is where we should add the rule.
